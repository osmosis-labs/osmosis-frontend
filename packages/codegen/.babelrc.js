const useESModules = !!process.env.MODULE;

module.exports = (api) => {
  api.cache(() => process.env.MODULE);
  return {
    plugins: [
      ['@babel/transform-runtime', { useESModules }],
      '@babel/proposal-object-rest-spread',
      '@babel/proposal-class-properties',
      '@babel/plugin-proposal-numeric-separator',
      '@babel/plugin-proposal-nullish-coalescing-operator',
      '@babel/plugin-proposal-optional-chaining',
      '@babel/proposal-export-default-from'
    ],
    presets: useESModules ? ['@babel/typescript'] : ['@babel/typescript', '@babel/env']
  };
};
