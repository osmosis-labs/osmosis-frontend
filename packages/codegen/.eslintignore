node_modules/
dist/
main/
module/
coverage/