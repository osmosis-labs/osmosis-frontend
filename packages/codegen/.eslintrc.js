module.exports = {
  plugins: ['prettier'],
  extends: ['eslint:recommended', 'prettier'],
  parser: '@babel/eslint-parser',
  parserOptions: {
    ecmaVersion: 11,
    requireConfigFile: false,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true
    }
  },
  env: {
    es6: true,
    browser: true,
    node: true,
    jest: true
  },
  rules: {
    'no-debugger': 2,
    'no-alert': 2,
    'no-await-in-loop': 0,
    'no-prototype-builtins': 0,
    'no-return-assign': ['error', 'except-parens'],
    'no-restricted-syntax': [
      2,
      'ForInStatement',
      'LabeledStatement',
      'WithStatement'
    ],
    'no-unused-vars': [
      0,
      {
        ignoreSiblings: true,
        argsIgnorePattern: 'React|res|next|^_'
      }
    ],
    'prefer-const': [
      'error',
      {
        destructuring: 'all'
      }
    ],
    'no-unused-expressions': [
      2,
      {
        allowTaggedTemplates: true
      }
    ],
    'no-console': 1,
    'comma-dangle': 2,
    'jsx-quotes': [2, 'prefer-double'],
    'linebreak-style': ['error', 'unix'],
    quotes: [
      2,
      'single',
      {
        avoidEscape: true,
        allowTemplateLiterals: true
      }
    ],
    'prettier/prettier': [
      'error',
      {
        trailingComma: 'none',
        singleQuote: true,
        printWidth: 80
      }
    ]
  }
};
