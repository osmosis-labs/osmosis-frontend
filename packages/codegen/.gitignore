# Logs
logs
*.log
npm-debug.log*

# Runtime data
pids
*.pid
*.seed

# dist
out
main
module

# Directory for instrumented libs generated by jscoverage/JSCover
lib-cov

# Coverage directory used by tools like istanbul
coverage

# nyc test coverage
.nyc_output

# Compiled binary addons (http://nodejs.org/api/addons.html)
build/Release

# Dependency directories
node_modules
jspm_packages

# Optional npm cache directory
.npm

# Optional REPL history
.node_repl_history

# Editors
.idea

# Lib
lib

# npm package lock
package-lock.json
yarn.lock

# others
.DS_Store