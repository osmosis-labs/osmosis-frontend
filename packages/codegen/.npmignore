*.log
npm-debug.log*

protobuf
out
*.proto

# snapshots
*.snap

# Coverage directory used by tools like istanbul
coverage
.nyc_output

# Dependency directories
node_modules

# npm package lock
package-lock.json
yarn.lock

# project files
__snapshots__
__fixtures__
__tests__
.babelrc
.babelrc.js
.editorconfig
.eslintignore
.eslintrc
.eslintrc.js
.gitignore
.travis.yml
.vscode
CHANGELOG.md
examples
jest.config.js
package.json
test
out
