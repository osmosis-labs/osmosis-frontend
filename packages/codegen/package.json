{
  "name": "@osmosis-labs/codegen",
  "version": "0.12.1",
  "description": "Typescript Protobuf Messages for Osmosis",
  "author": "Dan Lynch <pyramation@gmail.com>",
  "homepage": "https://github.com/osmosis-labs/osmosis-frontend/tree/master/packages/codegen#readme",
  "license": "Apache-2.0",
  "main": "main/index.js",
  "typings": "types/index.d.ts",
  "files": [
    "src",
    "main",
    "types"
  ],
  "scripts": {
    "build": "cross-env BABEL_ENV=production babel src --out-dir main --delete-dir-on-start --extensions \".tsx,.ts,.js\"",
    "build:ts": "tsc --project ./tsconfig.json",
    "codegen": "cross-env NODE_ENV=development babel-node scripts/codegen.js",
    "prepare": "npm run build",
    "lint": "eslint .",
    "format": "eslint . --fix",
    "clean": "rimraf ./types && rimraf ./src/codegen",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:debug": "node --inspect node_modules/.bin/jest --runInBand"
  },
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/osmosis-labs/osmosis-frontend"
  },
  "keywords": [],
  "bugs": {
    "url": "https://github.com/osmosis-labs/osmosis-frontend/issues"
  },
  "devDependencies": {
    "@babel/cli": "7.18.10",
    "@babel/core": "7.19.0",
    "@babel/eslint-parser": "^7.18.9",
    "@babel/generator": "7.19.0",
    "@babel/node": "^7.18.10",
    "@babel/parser": "^7.19.0",
    "@babel/plugin-proposal-class-properties": "7.18.6",
    "@babel/plugin-proposal-export-default-from": "7.18.10",
    "@babel/plugin-proposal-nullish-coalescing-operator": "7.18.6",
    "@babel/plugin-proposal-numeric-separator": "7.18.6",
    "@babel/plugin-proposal-object-rest-spread": "7.18.9",
    "@babel/plugin-proposal-optional-chaining": "7.18.9",
    "@babel/plugin-transform-runtime": "7.18.10",
    "@babel/preset-env": "7.19.0",
    "@babel/preset-typescript": "^7.17.12",
    "@babel/traverse": "7.19.0",
    "@babel/types": "7.19.0",
    "@osmonauts/telescope": "0.18.8",
    "@protobufs/cosmos": "^0.0.11",
    "@protobufs/cosmwasm": "^0.0.11",
    "@protobufs/ibc": "^0.0.11",
    "@types/jest": "^29.0.0",
    "babel-core": "7.0.0-bridge.0",
    "babel-jest": "29.0.2",
    "babel-watch": "^7.0.0",
    "case": "1.6.3",
    "cross-env": "^7.0.2",
    "eslint": "8.23.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-prettier": "^4.0.0",
    "glob": "8.0.3",
    "jest": "^29.0.2",
    "mkdirp": "1.0.4",
    "prettier": "^2.7.0",
    "regenerator-runtime": "^0.13.7",
    "rimraf": "3.0.2",
    "ts-jest": "^28.0.8",
    "typescript": "^4.8.2"
  },
  "dependencies": {
    "@babel/runtime": "^7.19.0",
    "@cosmjs/amino": "0.28.13",
    "@cosmjs/proto-signing": "0.28.13",
    "@cosmjs/stargate": "0.28.13",
    "@cosmjs/tendermint-rpc": "^0.28.13",
    "@osmonauts/helpers": "^0.4.3",
    "long": "^5.2.0",
    "protobufjs": "^6.11.3"
  }
}