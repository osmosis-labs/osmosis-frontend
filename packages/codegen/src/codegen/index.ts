/**
  * This file and any referenced files were automatically generated by @osmonauts/telescope@0.18.8
  * DO NOT MODIFY BY HAND. Instead, download the latest proto files for your chain
  * and run the transpile command or yarn proto command to regenerate this bundle.
  */
 
export * from "./ics23/bundle";
export * from "./cosmos_proto/bundle";
export * from "./cosmos/bundle";
export * from "./cosmos/client";
export * from "./cosmwasm/bundle";
export * from "./cosmwasm/client";
export * from "./gogoproto/bundle";
export * from "./google/bundle";
export * from "./ibc/bundle";
export * from "./ibc/client";
export * from "./osmosis/bundle";
export * from "./osmosis/client";
export * from "./tendermint/bundle";