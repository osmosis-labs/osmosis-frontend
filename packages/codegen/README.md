# @osmosis-labs/codegen

### Composing Messages

Import the `osmosis` object from `@osmosis-labs/codegen`. In this case, we're show the messages available from the `osmosis.gamm.v1beta1` module:

```js
import { osmosis } from 'osmojs';

const {
    joinPool,
    exitPool,
    exitSwapExternAmountOut,
    exitSwapShareAmountIn,
    joinSwapExternAmountIn,
    joinSwapShareAmountOut,
    swapExactAmountIn,
    swapExactAmountOut
} = osmosis.gamm.v1beta1.MessageComposer.withTypeUrl;
```

To see a complete list of messages, [see the section below](#osmosis-messages).

Now you can construct messages. If you use vscode or another typescript-enabled IDE, you should also be able to use `ctrl+space` to see auto-completion of the fields required for the message.

```js
import { coin } from '@cosmjs/amino';

const msg = swapExactAmountIn({
  sender,
  routes,
  tokenIn: coin(amount, denom),
  tokenOutMinAmount
});
```

### Osmosis Messages

```js
import { osmosis } from 'osmojs';

const {
    beginUnlocking,
    beginUnlockingAll,
    lockTokens
} = osmosis.lockup.MessageComposer.withTypeUrl;

const {
    lockAndSuperfluidDelegate,
    superfluidDelegate,
    superfluidUnbondLock,
    superfluidUndelegate
} = osmosis.superfluid.MessageComposer.withTypeUrl;

const {
    addToGauge,
    createGauge
} = osmosis.incentives.MessageComposer.withTypeUrl;

const {
    joinPool,
    exitPool,
    exitSwapExternAmountOut,
    exitSwapShareAmountIn,
    joinSwapExternAmountIn,
    joinSwapShareAmountOut,
    swapExactAmountIn,
    swapExactAmountOut
} = osmosis.gamm.v1beta1.MessageComposer.withTypeUrl;
```

### IBC Messages

```js
import { ibc } from '@osmosis-labs/codegen';

const {
    transfer
} = ibc.applications.transfer.v1.MessageComposer.withTypeUrl
```

### Cosmos Messages

```js
import { cosmos } from '@osmosis-labs/codegen';

const {
    fundCommunityPool,
    setWithdrawAddress,
    withdrawDelegatorReward,
    withdrawValidatorCommission
} = cosmos.distribution.v1beta1.MessageComposer.fromPartial;

const {
    multiSend,
    send
} = cosmos.bank.v1beta1.MessageComposer.fromPartial;

const {
    beginRedelegate,
    createValidator,
    delegate,
    editValidator,
    undelegate
} = cosmos.staking.v1beta1.MessageComposer.fromPartial;

const {
    deposit,
    submitProposal,
    vote,
    voteWeighted
} = cosmos.gov.v1beta1.MessageComposer.fromPartial;
```

### CosmWasm Messages

```js
import { cosmwasm } from "@osmosis-labs/codegen";

const {
    clearAdmin,
    executeContract,
    instantiateContract,
    migrateContract,
    storeCode,
    updateAdmin
} = cosmwasm.wasm.v1.MessageComposer.withTypeUrl;
```

## Credits

Code built with the help of these related projects:

* [@osmonauts/telescope](https://github.com/osmosis-labs/telescope) a "babel for the Cosmos", Telescope is a TypeScript Transpiler for Cosmos Protobufs.

## Disclaimer

AS DESCRIBED IN THE OSMOSIS LICENSES, THE SOFTWARE IS PROVIDED “AS IS”, AT YOUR OWN RISK, AND WITHOUT WARRANTIES OF ANY KIND.

No developer or entity involved in creating Telescope will be liable for any claims or damages whatsoever associated with your use, inability to use, or your interaction with other users of the Telescope code, including any direct, indirect, incidental, special, exemplary, punitive or consequential damages, or loss of profits, cryptocurrencies, tokens, or anything else of value.
