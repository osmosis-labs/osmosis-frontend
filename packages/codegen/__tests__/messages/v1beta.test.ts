
import { osmosis } from '../../src/codegen';

it('toJSON', () => {
    expect(Object.keys(osmosis.gamm.v1beta1.MessageComposer.toJSON)).toMatchSnapshot();
});

it('fromJSON', () => {
    expect(Object.keys(osmosis.gamm.v1beta1.MessageComposer.fromJSON)).toMatchSnapshot();
});

it('encoded', () => {
    expect(Object.keys(osmosis.gamm.v1beta1.MessageComposer.encoded)).toMatchSnapshot();
});
