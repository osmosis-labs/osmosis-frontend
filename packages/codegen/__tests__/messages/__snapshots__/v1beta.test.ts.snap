// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`encoded 1`] = `
[
  "joinPool",
  "exitPool",
  "swapExactAmountIn",
  "swapExactAmountOut",
  "joinSwapExternAmountIn",
  "joinSwapShareAmountOut",
  "exitSwapExternAmountOut",
  "exitSwapShareAmountIn",
]
`;

exports[`fromJSON 1`] = `
[
  "joinPool",
  "exitPool",
  "swapExactAmountIn",
  "swapExactAmountOut",
  "joinSwapExternAmountIn",
  "joinSwapShareAmountOut",
  "exitSwapExternAmountOut",
  "exitSwapShareAmountIn",
]
`;

exports[`toJSON 1`] = `
[
  "joinPool",
  "exitPool",
  "swapExactAmountIn",
  "swapExactAmountOut",
  "joinSwapExternAmountIn",
  "joinSwapShareAmountOut",
  "exitSwapExternAmountOut",
  "exitSwapShareAmountIn",
]
`;
